design/one-pager-stacks-security-isolation.md

```diff
@@ -0,0 +1,231 @@
+
+1. Stack `foo-v1.0` is installed along with its CRD that has version `v1beta1`
+1. A minor update to stack foo is released as version `foo-v1.1`
+    1. This update to the stack adds a new field to the CRD but does **not** increment the CRD version
+1. The control plane now has 2 versions of the stack's controllers (`foo-v1.0` and `foo-v1.1`), but only 1 version of the CRD (`v1beta`)
+
+Which CRD schema should "win" in the control plane?
+
+* The **new CRD schema** with the new field that the **old controller** doesn't understand?
+* The **old CRD schema** without the new field that the **new controller** wants to use?
+
+This speaks to the importance of properly versioning CRDs when they have schema changes.
+Kubernetes now supports [multiple side by side versions of CRDs](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definition-versioning/#specify-multiple-versions) in the cluster, so this is actually possible in more recent versions of Kubernetes.
+We suspect (and hope) that this newer CRD versions functionality will encourage CRD authors to adopt a more strict process of updating their CRD versions, which would avoid the issue raised in this section.
+
+## Permissions and Privileges
+
+### Stack Manager
+
+Currently, the Stack Manager that installs stacks into the cluster runs with a `ClusterRoleBinding` to the cluster-admin `ClusterRole` because it needs to create RBAC permissions for the arbitrary owned and dependent CRDs of arbitrary stacks.
+A service account cannot grant RBAC permissions that it doesn't already have itself, so to support the arbitrary permissions needs of all stacks, we are currently using the cluster-admin role.
+
+This is a similar problem to what Helm's Tiller has historically had and we can consider revisiting this in the future.
+One possible solution would be to do more client side processing of stacks (instead of in-cluster processing) using the credentials and permissions of the interactive user,
+which would be similar to [Helm 3's](https://github.com/helm/community/blob/master/helm-v3/000-helm-v3.md) major design change.
+
+However, because we are limiting the [allowed set of resources](#allowed-resource-access) that a stack can have access to, this in-cluster Stack Manager model may be acceptable.
+Furthermore, moving to a client side model would limit the full lifecycle management abilities available from an in-cluster Stack Manager, such as automatic stack upgrades.
+
+A full exploration of this topic is out of scope for this version of the design document.
+
+### Stack Package Author
+
+The author of a stack will need to specify the intended scope of permissions for their stack.
+This should be known at stack build time because the stack's controller(s) will need to know whether to watch resources in a single namespace or across all namespaces.
+
+The `app.yaml` schema should be updated with a new field that captures the stack author's intended permission scope:
```

Comment:
Did we already consider using a `kind` in the `app.yaml` for this? Similar to how the installation types are being explained

Reply:
no i had not considered using a `Kind` field, but it could potentially make sense instead of an enum.  I'm not 100% either way yet.